crates/services/gas_price_service/src/fuel_gas_price_updater/fuel_da_source_adapter/block_committer_costs.rs

```diff
@@ -0,0 +1,60 @@
+use crate::fuel_gas_price_updater::{
+    service::Result as DaBlockCostsResult,
+    DaBlockCosts,
+    DaBlockCostsSource,
+};
+use anyhow::anyhow;
+use reqwest::Url;
+use serde::{
+    Deserialize,
+    Serialize,
+};
+
+/// This struct is used to denote the block committer da gas price source,,
+/// which receives data from the block committer (only http api for now)
+pub struct BlockCommitterDaBlockCosts {
+    client: reqwest::Client,
+    url: Url,
+}
+
+#[derive(Debug, Deserialize, Serialize, Clone, Default, PartialEq)]
+struct RawDaBlockCosts {
+    pub l2_block_range: core::ops::Range<u32>,
+    pub blob_size_bytes: u32,
```

Comment:
the blob size will always be < 128kb

Reply:
The block committer will return cumulative value, not the blob size of the last committed blob.

Reply:
don't think so, we discussed with @MujkicA and I don't think it will return cumulative value

Reply:
I don't see how they can return anything else with the current API. Either they return cumulative values, or the committer provides another API like `get_cost_at_l1_height(block_height: DaBlockHeight)` or `get_cost_at_l2_height(block_height: BlockHeight)`(or instead of the height it uses range).


Because the committer can't know for what range it should return data based on the current API.

Reply:
the block committer will create a new api for us :)

Reply:
If they plan to create a new API, then we need to agree on it in advance and implement logic based on this agreement=)

But another question: do we need a new API? It seems for us is enough to receive cumulative values with latest L1 and L2 block heights and it should fulfill our requirements

Reply:
Since there are two layers of abstraction between a block and what is actually posted inside blobs, we can only provide a cumulative value over some block range. We can tailor the api to your needs given that constraint.

See other [comment](https://github.com/FuelLabs/fuel-core/pull/2163#discussion_r1750668108)

Reply:
From my perspective, it should be simpler just to use cumulative value for all time=) But it is up to @MitchTurner  and @rymnc to use a small range or the whole range. Based on the definition of `RawDaBlockCosts,` I see that we don't have all the information we need right now. We also need to know the DA block height of latest committed blob